src/app/ember_coupling/EventPathValidity.mixin.h

```diff
@@ -0,0 +1,155 @@
+/*
+ *
+ *    Copyright (c) 2020-2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+// This is intended as a TOP LEVEL INCLUDE inside code that uses it
+
+namespace chip {
+namespace app {
+namespace {
+
+/**
+ * Find out if the given EventId is reported as supported by the given cluster
+ * within its metadata. If cluster has no event metadata (i.e. no event list
+ * support is available), clusters are assumed to support any event as there is
+ * no way to actually tell.
```

Comment:
What if I don't have a partial event list in ZAP, but I actually emit more events in code. Is that a problem?

Reply:
Current logic is that we just allow any event to be emitted, it does not matter if it is in zap list or not.

Reply:
Resolving this: current behavior is currently preserved and we actually have no validation that we only emit events in the list.

Long term it seems we may plan to migrate away from having the list in the first place as it cannot really be validated/enforced. EventList is provisional already.